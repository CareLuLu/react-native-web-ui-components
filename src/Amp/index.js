import React, { useContext } from 'react';

const AMP = React.createContext('amp');

export const { Provider } = AMP;

export const { Consumer } = AMP;

export const withAmp = () => Component => props => (
  <AMP.Consumer>
    {amp => <Component {...props} amp={amp} />}
  </AMP.Consumer>
);

export const useAmp = () => useContext(AMP);

export default {
  Provider,
  Consumer,
  useAmp,
  withAmp,
};
