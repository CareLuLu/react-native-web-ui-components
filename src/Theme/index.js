import React, { useContext } from 'react';
import { StyleSheet, Platform } from 'react-native';
import omit from 'lodash/omit';
import PropTypes from 'prop-types';
import merge from 'lodash/merge';

const Theme = React.createContext('theme');

const themeProps = [
  'fontFamily',
  'theme',
  'themeTextStyle',
  'themeInputStyle',
  'themePrimaryStyle',
];

const defaults = {
  canonical: uri => uri.replace('/amp'),
  resource: uri => uri,
  omit: props => omit(props, themeProps),
  '*': {
    fontFamily: {
      regular: 'Lucida Sans',
      bold: 'Lucida Sans Bold',
    },
  },
  Link: {
    basepath: 'localhost',
  },
  Datepicker: {
    selectedDateColor: '#337AB7',
  },
  input: {
    regular: StyleSheet.create({
      background: { backgroundColor: '#FFFFFF' },
      border: {
        borderWidth: 1,
        borderStyle: 'solid',
        borderColor: '#D3D6D6',
        borderRadius: 2,
      },
      text: { color: '#545454' },
      placeholder: { color: '#D3D6D6' },
      opacity: { opacity: 1 },
      selected: { color: '#0E73CA' },
      unselected: { color: '#BDC3C7' },
    }),
    readonly: StyleSheet.create({
      background: { backgroundColor: '#F4F6F6' },
      border: {
        borderWidth: 1,
        borderStyle: 'solid',
        borderColor: '#D5DBDB',
        borderRadius: 2,
      },
      text: { color: '#545454' },
      placeholder: { color: '#D3D6D6' },
      opacity: { opacity: 0.7 },
      selected: { color: '#0E73CA' },
      unselected: { color: '#BDC3C7' },
    }),
    disabled: StyleSheet.create({
      background: { backgroundColor: '#F4F6F6' },
      border: {
        borderWidth: 1,
        borderStyle: 'solid',
        borderColor: '#D5DBDB',
        borderRadius: 2,
      },
      text: { color: '#545454' },
      placeholder: { color: '#D3D6D6' },
      opacity: { opacity: 0.7 },
      selected: { color: '#0E73CA' },
      unselected: { color: '#BDC3C7' },
    }),
    focused: StyleSheet.create({
      background: { backgroundColor: '#FFFFFF' },
      border: {
        borderWidth: 1,
        borderStyle: 'solid',
        borderColor: '#6CB3FF',
        borderRadius: 2,
      },
      text: { color: '#545454' },
      placeholder: { color: '#D3D6D6' },
      opacity: { opacity: 1 },
      selected: { color: '#0E73CA' },
      unselected: { color: '#BDC3C7' },
    }),
    error: StyleSheet.create({
      background: { backgroundColor: '#FFFFFF' },
      border: {
        borderWidth: 1,
        borderStyle: 'solid',
        borderColor: '#EE2D68',
        borderRadius: 2,
      },
      text: { color: '#545454' },
      placeholder: { color: '#EE2D68' },
      opacity: { opacity: 1 },
      selected: { color: '#0E73CA' },
      unselected: { color: '#BDC3C7' },
    }),
  },
  colors: {
    text: 'gray',
    primary: 'navy',
    pink: StyleSheet.create({
      background: { backgroundColor: '#F15786' },
      border: { borderColor: '#BF4E71' },
      text: { color: '#F15786' },
    }),
    yellow: StyleSheet.create({
      background: { backgroundColor: '#FEB715' },
      border: { borderColor: '#D99D14' },
      text: { color: '#FEB715' },
    }),
    gray: StyleSheet.create({
      background: { backgroundColor: '#D4D4D4' },
      border: { borderColor: '#BFBFBF' },
      text: { color: '#545454' },
    }),
    lightGray: StyleSheet.create({
      background: { backgroundColor: '#BFBFBF' },
      border: { borderColor: '#BFBFBF' },
      text: { color: '#BFBFBF' },
    }),
    white: StyleSheet.create({
      background: { backgroundColor: '#FFFFFF' },
      border: { borderColor: '#BFBFBF' },
      text: { color: '#FFFFFF' },
    }),
    navy: StyleSheet.create({
      background: { backgroundColor: '#0E73CA' },
      border: { borderColor: '#055396' },
      text: { color: '#0E73CA' },
    }),
    teal: StyleSheet.create({
      background: { backgroundColor: '#23AAAA' },
      border: { borderColor: '#23AAAA' },
      text: { color: '#23AAAA' },
    }),
    black: StyleSheet.create({
      background: { backgroundColor: '#000000' },
      border: { borderColor: '#000000' },
      text: { color: '#000000' },
    }),
  },
  platform: {
    web: {
      '*': {
        fontFamily: {
          regular: '"Lucida Sans Unicode","Lucida Grande",Arial,Helvetica,clean,sans-serif',
          bold: '"Lucida Grande", "Lucida Sans Unicode","Lucida Grande",Arial,Helvetica,clean,sans-serif',
        },
      },
    },
  },
};

const getInputStyle = (theme, {
  disabled,
  readonly,
  hasError,
  autoFocus,
}) => {
  let label = 'regular';
  if (hasError) {
    label = 'error';
  } else if (disabled) {
    label = 'disabled';
  } else if (readonly) {
    label = 'readonly';
  } else if (autoFocus) {
    label = 'focused';
  }
  return theme.input[label];
};

export class Provider extends React.Component {
  static propTypes = {
    value: PropTypes.shape(),
  };

  static defaultProps = {
    value: {},
  };

  static getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.value !== prevState.value) {
      return { value: merge({}, defaults, nextProps.value) };
    }
    return null;
  }

  constructor(props) {
    super(props);
    const { value } = props;
    this.state = { value: merge({}, defaults, value) };
  }

  render() {
    const { value } = this.state;
    return <Theme.Provider {...this.props} value={value} />;
  }
}

export const { Consumer } = Theme;

export const useTheme = (type, { style, ...props }) => {
  const theme = useContext(Theme);
  return {
    ...(theme['*'] || {}),
    ...((theme.platform[Platform.OS] && theme.platform[Platform.OS]['*']) || {}),
    ...(theme[type] || {}),
    ...((theme.platform[Platform.OS] && theme.platform[Platform.OS][type]) || {}),
    ...props,
    style: [theme[type] && theme[type].style, style],
    theme,
    themeTextStyle: theme.colors[theme.colors.text],
    themePrimaryStyle: theme.colors[theme.colors.primary],
    themeInputStyle: getInputStyle(theme, props),
  };
};

export const withTheme = type => Component => ({ style, ...props }) => { // eslint-disable-line
  Component.displayName = type; // eslint-disable-line
  return (
    <Theme.Consumer>
      {theme => (
        <Component
          {...(theme['*'] || {})}
          {...((theme.platform[Platform.OS] && theme.platform[Platform.OS]['*']) || {})}
          {...(theme[type] || {})}
          {...((theme.platform[Platform.OS] && theme.platform[Platform.OS][type]) || {})}
          {...props}
          style={[theme[type] && theme[type].style, style]}
          theme={theme}
          themeTextStyle={theme.colors[theme.colors.text]}
          themePrimaryStyle={theme.colors[theme.colors.primary]}
          themeInputStyle={getInputStyle(theme, props)}
        />
      )}
    </Theme.Consumer>
  );
};

export default {
  withTheme,
  Provider,
  Consumer,
};
