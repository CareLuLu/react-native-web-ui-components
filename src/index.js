export { default as Alert } from './Alert';
export { default as Amp } from './Amp';
export { default as Banner } from './Banner';
export { default as Bold } from './Bold';
export { default as Box } from './Box';
export { default as BoxButtonRow } from './BoxButtonRow';
export { default as BoxHeader } from './BoxHeader';
export { default as BoxItem } from './BoxItem';
export { default as BoxTitle } from './BoxTitle';
export { default as Button } from './Button';
export { default as Carousel } from './Carousel';
export { default as Center } from './Center';
export { default as Checkbox } from './Checkbox';
export { default as Column } from './Column';
export { default as Confirm } from './Confirm';
export { default as Container } from './Container';
export { default as createDomStyle } from './createDomStyle';
export { default as Datepicker } from './Datepicker';
export { default as Draggable } from './Draggable';
export { default as Dropzone } from './Dropzone';
export { default as Helmet } from './Helmet';
export { default as HideShowText } from './HideShowText';
export { default as Hr } from './Hr';
export { default as Icon } from './Icon';
export { default as IconLink } from './IconLink';
export { default as IconText } from './IconText';
export { default as Image } from './Image';
export { default as Link } from './Link';
export { default as Loading } from './Loading';
export { default as MainContainer } from './MainContainer';
export { default as Modal } from './Modal';
export { default as ModalPicker } from './ModalPicker';
export { default as NavLink } from './NavLink';
export { default as Popup } from './Popup';
export { default as Radiobox } from './Radiobox';
export { default as Rating } from './Rating';
export { default as ReadMoreLessLink } from './ReadMoreLessLink';
export { default as Router } from './Router';
export { default as Row } from './Row';
export { default as Screen } from './Screen';
export { default as ScrollView } from './ScrollView';
export { default as Select } from './Select';
export { default as Sidebar } from './Sidebar';
export { default as Slider } from './Slider';
export { default as Spinner } from './Spinner';
export { default as StylePropType } from './StylePropType';
export { default as Switch } from './Switch';
export { default as Text } from './Text';
export { default as TextInput } from './TextInput';
export { default as Theme } from './Theme';
export { default as TimeRangePicker } from './TimeRangePicker';
export { default as Title } from './Title';
export { default as Tooltip } from './Tooltip';
export { default as TouchableOpacity } from './TouchableOpacity';
export { default as TouchableWithoutFeedback } from './TouchableWithoutFeedback';
export { default as UIProvider } from './UIProvider';
export { default as View } from './View';
export { default as WebOnly } from './WebOnly';
export { default as WebView } from './WebView';
