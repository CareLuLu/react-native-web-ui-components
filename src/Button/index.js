import React from 'react';
import PropTypes from 'prop-types';
import noop from 'lodash/noop';
import { StyleSheet } from 'react-native';
import StylePropType from '../StylePropType';
import { withTheme } from '../Theme';
import Text from '../Text';
import Link from '../Link';
import View from '../View';
import TouchableOpacity from '../TouchableOpacity';

const styles = StyleSheet.create({
  empty: {},
  auto: {
    alignSelf: 'flex-start',
  },
  fullWidth: {
    width: '100%',
  },
  outer: {
    marginRight: 5,
    marginBottom: 5,
    borderBottomWidth: 2,
  },
  whiteOuter: {
    borderBottomWidth: 1,
    borderWidth: 1,
  },
  flat: {
    borderBottomWidth: 0,
    borderWidth: 0,
  },
  radius: {
    borderRadius: 2,
  },
  text: {
    textAlign: 'center',
    fontSize: 14,
    paddingTop: 7,
    paddingBottom: 7,
    paddingLeft: 20,
    paddingRight: 20,
  },
  small: {
    fontSize: 13,
    paddingTop: 5,
    paddingBottom: 5,
    paddingLeft: 10,
    paddingRight: 10,
  },
  extraSmall: {
    fontSize: 11,
    paddingTop: 2,
    paddingBottom: 2,
    paddingLeft: 7,
    paddingRight: 7,
  },
  nomargin: {
    marginRight: 0,
    marginBottom: 0,
  },
});

const getTextType = (textType, type) => {
  if (textType) {
    return textType;
  }
  if (type === 'blue') {
    return 'blue';
  }
  if (type === 'white') {
    return 'gray';
  }
  return 'white';
};

const ifProp = (props, propName) => {
  if (props[propName]) {
    return styles[propName];
  }
  return styles.empty;
};

const Button = (props) => {
  const {
    to,
    auto,
    type,
    style,
    textStyle,
    blank,
    onPress,
    replace,
    children,
    textType,
    theme,
    className,
  } = props;
  let width = null;
  let height = null;
  let isAuto = auto;
  let buttonWidth = isAuto ? 'button-auto' : 'button-full';
  const css = StyleSheet.flatten(style || {});
  if (css.width !== undefined) {
    isAuto = true;
    buttonWidth = 'button-auto';
    width = { width: css.width };
  }
  if (css.height !== undefined) {
    height = { height: css.height };
  }
  return (
    <TouchableOpacity
      className={`${className} button-wrapper ${buttonWidth}`}
      onPress={onPress}
      style={[isAuto ? styles.empty : styles.fullWidth, style]}
    >
      <View
        className={`button button-${type}`}
        style={[
          styles.outer,
          theme.colors[type] && theme.colors[type].border,
          theme.colors[type] && theme.colors[type].background,
          styles[`${type}Outer`],
          isAuto ? styles.auto : styles.empty,
          ifProp(props, 'flat'),
          ifProp(props, 'radius'),
          ifProp(props, 'nomargin'),
          width,
          height,
        ]}
      >
        {to ? (
          <Link
            replace={replace}
            type={getTextType(textType, type)}
            blank={blank}
            to={to}
            style={[styles.text, ifProp(props, 'small'), ifProp(props, 'extraSmall'), textStyle]}
          >
            {children}
          </Link>
        ) : (
          <Text
            type={getTextType(textType, type)}
            style={[styles.text, ifProp(props, 'small'), ifProp(props, 'extraSmall'), textStyle]}
          >
            {children}
          </Text>
        )}
      </View>
    </TouchableOpacity>
  );
};

Button.propTypes = {
  theme: PropTypes.shape().isRequired,
  type: PropTypes.string.isRequired,
  onPress: PropTypes.func,
  children: PropTypes.node,
  to: PropTypes.string,
  blank: PropTypes.bool,
  textType: PropTypes.string,
  style: StylePropType,
  textStyle: StylePropType,
  auto: PropTypes.bool,
  className: PropTypes.string,
  replace: PropTypes.bool,
  small: PropTypes.bool, // eslint-disable-line
  extraSmall: PropTypes.bool, // eslint-disable-line
  flat: PropTypes.bool,  // eslint-disable-line
  radius: PropTypes.bool,  // eslint-disable-line
  nomargin: PropTypes.bool,  // eslint-disable-line
};

Button.defaultProps = {
  children: null,
  small: false,
  extraSmall: false,
  flat: true,
  radius: false,
  onPress: noop,
  to: null,
  blank: false,
  textType: null,
  style: null,
  textStyle: null,
  nomargin: false,
  auto: false,
  className: '',
  replace: false,
};

export default withTheme('Button')(Button);
