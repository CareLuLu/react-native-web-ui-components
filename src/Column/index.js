import React from 'react';
import PropTypes from 'prop-types';
import {
  StyleSheet,
  View,
  ViewPropTypes,
} from 'react-native';
import { useScreen } from '../Screen';
import { withTheme } from '../Theme';
import { pick } from '../utils';

const styles = StyleSheet.create({
  empty: {},
  defaults: {
    flexDirection: 'column',
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
  },
  width1: { width: '8.33333333%' },
  width2: { width: '16.66666667%' },
  width3: { width: '25%' },
  width4: { width: '33.33333333%' },
  width5: { width: '41.66666667%' },
  width6: { width: '50%' },
  width7: { width: '58.33333333%' },
  width8: { width: '66.66666667%' },
  width9: { width: '75%' },
  width10: { width: '83.33333333%' },
  width11: { width: '91.66666667%' },
  width12: { width: '100%' },
});

const Column = ({
  lg,
  md,
  sm,
  xs,
  style,
  absolute,
  children,
}) => {
  const screen = useScreen();

  let width;
  switch (screen.type) {
    case 'lg': width = pick(lg, md, sm, xs); break;
    case 'md': width = pick(md, sm, xs); break;
    case 'sm': width = pick(sm, xs); break;
    default: width = xs;
  }
  if (width === 0) {
    return null;
  }
  let columnStyle;
  if (width === null) {
    if (style === styles.empty) {
      columnStyle = styles.defaults;
    } else {
      columnStyle = [styles.defaults, style];
    }
  } else if (!absolute) {
    columnStyle = [styles.defaults, style, styles[`width${width}`]];
  } else {
    columnStyle = [styles.defaults, style, { width }];
  }
  return (
    <View style={columnStyle}>
      {children}
    </View>
  );
};

Column.propTypes = {
  children: PropTypes.node,
  style: ViewPropTypes.style,
  xs: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  sm: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  md: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  lg: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
  absolute: PropTypes.bool,
};

Column.defaultProps = {
  children: null,
  style: styles.empty,
  xs: null,
  sm: null,
  md: null,
  lg: null,
  absolute: false,
};

export default withTheme('Column')(Column);
