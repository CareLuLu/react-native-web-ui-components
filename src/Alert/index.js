import React, { useState, useCallback } from 'react';
import PropTypes from 'prop-types';
import { StyleSheet } from 'react-native';
import noop from 'lodash/noop';
import { withTheme } from '../Theme';
import Popup from '../Popup';
import Button from '../Button';
import Center from '../Center';
import Text from '../Text';

const styles = StyleSheet.create({
  text: {
    textAlign: 'center',
    paddingTop: 5,
    paddingBottom: 15,
  },
});

const Alert = ({
  theme,
  children,
  visible,
  onOk,
  ...props
}) => {
  const [isVisible, setVisible] = useState(visible);
  const hide = useCallback(() => {
    setVisible(false);
    onOk();
  }, [setVisible, onOk]);

  return (
    <Popup
      shouldCloseOnEsc
      {...props}
      onClose={onOk}
      followKeyboard
      visible={isVisible}
      className="NoAdjustment"
    >
      {typeof children === 'string' ? (
        <Text style={styles.text}>{children}</Text>
      ) : children}
      <Center>
        <Button auto type={theme.colors.primary} flat={false} onPress={hide}>Ok</Button>
      </Center>
    </Popup>
  );
};

Alert.propTypes = {
  theme: PropTypes.shape().isRequired,
  visible: PropTypes.bool,
  children: PropTypes.node,
  onOk: PropTypes.func,
};

Alert.defaultProps = {
  visible: true,
  children: null,
  onOk: noop,
};

export default withTheme('Alert')(Alert);
