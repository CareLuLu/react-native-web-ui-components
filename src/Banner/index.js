import React from 'react';
import PropTypes from 'prop-types';
import { StyleSheet, ImageBackground } from 'react-native';
import StylePropType from '../StylePropType';
import { withTheme } from '../Theme';
import { useScreen } from '../Screen';
import Column from '../Column';

const styles = StyleSheet.create({
  empty: {},
  image: {
    width: '100%',
    height: '100%',
    flex: 1,
  },
});

const Banner = ({
  theme,
  columnXs,
  columnSm,
  columnMd,
  columnLg,
  height,
  fit,
  maxHeight,
  children,
  lg,
  md,
  sm,
  xs,
  style,
}) => {
  const screen = useScreen();
  const defaults = { height };
  if (fit) {
    defaults.height = screen.height;
  }
  if (defaults.height > maxHeight) {
    defaults.height = maxHeight;
  }
  let uri;
  switch (screen.type) {
    case 'lg': uri = lg || md || sm || xs; break;
    case 'md': uri = md || sm || xs; break;
    case 'sm': uri = sm || xs; break;
    default: uri = xs;
  }
  uri = theme.resource(uri, { maxWidth: screen.width, maxHeight: defaults.height });
  return (
    <Column
      xs={columnXs}
      sm={columnSm}
      md={columnMd}
      lg={columnLg}
      style={defaults}
    >
      <ImageBackground resizeMode="cover" source={{ uri }} style={[styles.image, style]}>
        {children}
      </ImageBackground>
    </Column>
  );
};

Banner.propTypes = {
  theme: PropTypes.shape({
    resource: PropTypes.func.isRequired,
  }).isRequired,
  children: PropTypes.node,
  fit: PropTypes.bool,
  height: PropTypes.number,
  maxHeight: PropTypes.number,
  xs: PropTypes.string,
  sm: PropTypes.string,
  md: PropTypes.string,
  lg: PropTypes.string,
  columnXs: PropTypes.number,
  columnSm: PropTypes.number,
  columnMd: PropTypes.number,
  columnLg: PropTypes.number,
  style: StylePropType,
};

Banner.defaultProps = {
  children: null,
  fit: true,
  height: null,
  maxHeight: 700,
  xs: null,
  sm: null,
  md: null,
  lg: null,
  columnXs: 12,
  columnSm: null,
  columnMd: null,
  columnLg: null,
  style: styles.empty,
};

export default withTheme('Banner')(Banner);
