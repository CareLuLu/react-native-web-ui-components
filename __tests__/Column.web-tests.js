import React from 'react';
import renderer from 'react-test-renderer';
import { StyleSheet } from 'react-native';
import Column from '../src/Column/index.web';
import { Provider as ThemeProvider } from '../src/Theme';

let testDescription = '';
let testData = '';

testDescription = 'Column for Web should render View with class = ` col-xs-9 col-sm-10 col-md-11 col-lg-12`.\n';
testData = 'Test Data: lg={12} md={11} sm={10} xs={9}';
test(testDescription + testData, () => {
  const tree = renderer.create(
    <ThemeProvider>
      <Column lg={12} md={11} sm={10} xs={9} />
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

testDescription = 'Column for Web should render View with class = ` col-xs-9 col-sm-10 col-md-11 col-lg-11`.\n';
testData = 'Test Data: md={11} sm={10} xs={9}';
test(testDescription + testData, () => {
  const tree = renderer.create(
    <ThemeProvider>
      <Column md={11} sm={10} xs={9} />
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

testDescription = 'Column for Web should render View with class = ` col-xs-9 col-sm-10 col-md-10 col-lg-10`.\n';
testData = 'Test Data: sm={10} xs={9}';
test(testDescription + testData, () => {
  const tree = renderer.create(
    <ThemeProvider>
      <Column sm={10} xs={9} />
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

testDescription = 'Column for Web should render View with class = ` col-xs-9 col-sm-9 col-md-9 col-lg-9`.\n';
testData = 'Test Data: xs={9}';
test(testDescription + testData, () => {
  const tree = renderer.create(
    <ThemeProvider>
      <Column xs={9} />
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

testDescription = 'Column for Web should render View with class = ` col-lg-10 col-md-10 col-sm-10`.\n'
                    + 'The following classes should not be displayed: col-xs.\n';
testData = 'Test Data: sm={10}';
test(testDescription + testData, () => {
  const tree = renderer.create(
    <ThemeProvider>
      <Column sm={10} />
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

testDescription = 'Column for Web should render View with class = ` col-lg-11 col-md-11`.\n'
                    + 'The following classes should not be displayed: col-sm col-xs.\n';
testData = 'Test Data: md={11}';
test(testDescription + testData, () => {
  const tree = renderer.create(
    <ThemeProvider>
      <Column md={11} />
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

testDescription = 'Column for Web should render View with class = ` col-lg-12`.\n'
                    + 'The following classes should not be displayed: col-md col-sm col-xs.\n';
testData = 'Test Data: lg={12}';
test(testDescription + testData, () => {
  const tree = renderer.create(
    <ThemeProvider>
      <Column lg={12} />
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

testDescription = 'Column for Web should render View with class = ` col-xs-9 col-sm-10 col-md-11 col-lg-12` and "width": 9.\n';
testData = 'Test Data: lg={12} md={11} sm={10} xs={9} absolute={true}';
test(testDescription + testData, () => {
  const tree = renderer.create(
    <ThemeProvider>
      <Column lg={12} md={11} sm={10} xs={9} absolute />
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

testDescription = 'Column for Web should render View with class = ` col-xs-9 col-sm-10 col-md-11 col-lg-12` and "margin": 1.\n';
testData = 'Test Data: lg={12} md={11} sm={10} xs={9} absolute={true} "margin": 1';
test(testDescription + testData, () => {
  const styles = StyleSheet.create({
    margin: 1,
  });

  const tree = renderer.create(
    <ThemeProvider>
      <Column lg={12} md={11} sm={10} xs={9} style={styles} />
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

testDescription = 'Column for Web should render View with outer and inner column (parent and child).\n'
                    + 'Outer column (parent) should have class = ` col-xs-9 col-sm-10 col-md-11 col-lg-12`, "width": 9, and "margin": 1\n'
                    + 'Inner column (child) should have class = ` col-xs-1 col-sm-2 col-md-3 col-lg-4`, and "margin": 0.2\n';
testData = 'Test Data: Outer_Column { lg={12} md={11} sm={10} xs={9} absolute={true} "margin": 1}\n'
                + 'Inner_Column { lg={4} md={3} sm={2} xs={1} absolute={false} "margin": 0.2}';
test(testDescription + testData, () => {
  const stylesOfOuterColumn = StyleSheet.create({
    margin: 1,
  });

  const stylesOfInnerColumn = StyleSheet.create({
    margin: 0.2,
  });

  const tree = renderer.create(
    <ThemeProvider>
      <Column lg={12} md={11} sm={10} xs={9} style={stylesOfOuterColumn} absolute>
        <Column lg={4} md={3} sm={2} xs={1} style={stylesOfInnerColumn} absolute={false} />
      </Column>
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

testDescription = 'Column for Web should render View with outer and inner column (parent and child).\n'
                    + 'Outer column (parent) should have class = ` col-xs-9 col-sm-10 col-md-11 col-lg-12`, and "margin": 1.\n'
                    + 'Inner column (child) should have class = ` col-xs-1 col-sm-2 col-md-3 col-lg-4`, "width": 1, and "margin": 0.2\n';
testData = 'Test Data: Outer_Column { lg={12} md={11} sm={10} xs={9} absolute={false} "margin": 1}\n'
             + 'Inner_Column { lg={4} md={3} sm={2} xs={1} absolute={true} "margin": 0.2}';
test(testDescription + testData, () => {
  const stylesOfOuterColumn = StyleSheet.create({
    margin: 1,
  });

  const stylesOfInnerColumn = StyleSheet.create({
    margin: 0.2,
  });

  const tree = renderer.create(
    <ThemeProvider>
      <Column lg={12} md={11} sm={10} xs={9} style={stylesOfOuterColumn} absolute={false}>
        <Column lg={4} md={3} sm={2} xs={1} style={stylesOfInnerColumn} absolute />
      </Column>
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

testDescription = 'Column for Web should render View with class = ``.\n'
                    + 'The following classes should not be displayed: col-xs col-sm col-md col-lg\n';
testData = 'Test Data: xs=null sm=null md=null lg=null';
test(testDescription + testData, () => {
  const tree = renderer.create(
    <ThemeProvider>
      <Column />
    </ThemeProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
