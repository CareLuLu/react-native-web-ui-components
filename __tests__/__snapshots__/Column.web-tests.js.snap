// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column for Web should render View with class = \` col-lg-10 col-md-10 col-sm-10\`.
The following classes should not be displayed: col-xs.
Test Data: sm={10} 1`] = `
<View
  data-class=" col-sm-10 col-md-10 col-lg-10"
  style={
    Array [
      undefined,
      Array [
        undefined,
        undefined,
      ],
    ]
  }
/>
`;

exports[`Column for Web should render View with class = \` col-lg-11 col-md-11\`.
The following classes should not be displayed: col-sm col-xs.
Test Data: md={11} 1`] = `
<View
  data-class=" col-md-11 col-lg-11"
  style={
    Array [
      undefined,
      Array [
        undefined,
        undefined,
      ],
    ]
  }
/>
`;

exports[`Column for Web should render View with class = \` col-lg-12\`.
The following classes should not be displayed: col-md col-sm col-xs.
Test Data: lg={12} 1`] = `
<View
  data-class=" col-lg-12"
  style={
    Array [
      undefined,
      Array [
        undefined,
        undefined,
      ],
    ]
  }
/>
`;

exports[`Column for Web should render View with class = \` col-xs-9 col-sm-9 col-md-9 col-lg-9\`.
Test Data: xs={9} 1`] = `
<View
  data-class=" col-xs-9 col-sm-9 col-md-9 col-lg-9"
  style={
    Array [
      undefined,
      Array [
        undefined,
        undefined,
      ],
    ]
  }
/>
`;

exports[`Column for Web should render View with class = \` col-xs-9 col-sm-10 col-md-10 col-lg-10\`.
Test Data: sm={10} xs={9} 1`] = `
<View
  data-class=" col-xs-9 col-sm-10 col-md-10 col-lg-10"
  style={
    Array [
      undefined,
      Array [
        undefined,
        undefined,
      ],
    ]
  }
/>
`;

exports[`Column for Web should render View with class = \` col-xs-9 col-sm-10 col-md-11 col-lg-11\`.
Test Data: md={11} sm={10} xs={9} 1`] = `
<View
  data-class=" col-xs-9 col-sm-10 col-md-11 col-lg-11"
  style={
    Array [
      undefined,
      Array [
        undefined,
        undefined,
      ],
    ]
  }
/>
`;

exports[`Column for Web should render View with class = \` col-xs-9 col-sm-10 col-md-11 col-lg-12\` and "margin": 1.
Test Data: lg={12} md={11} sm={10} xs={9} absolute={true} "margin": 1 1`] = `
<View
  data-class=" col-xs-9 col-sm-10 col-md-11 col-lg-12"
  style={
    Array [
      undefined,
      Array [
        undefined,
        Object {
          "margin": 1,
        },
      ],
    ]
  }
/>
`;

exports[`Column for Web should render View with class = \` col-xs-9 col-sm-10 col-md-11 col-lg-12\` and "width": 9.
Test Data: lg={12} md={11} sm={10} xs={9} absolute={true} 1`] = `
<View
  data-class=" col-xs-9 col-sm-10 col-md-11 col-lg-12"
  style={
    Array [
      undefined,
      Array [
        Array [
          undefined,
          undefined,
        ],
        Object {
          "width": 9,
        },
      ],
    ]
  }
/>
`;

exports[`Column for Web should render View with class = \` col-xs-9 col-sm-10 col-md-11 col-lg-12\`.
Test Data: lg={12} md={11} sm={10} xs={9} 1`] = `
<View
  data-class=" col-xs-9 col-sm-10 col-md-11 col-lg-12"
  style={
    Array [
      undefined,
      Array [
        undefined,
        undefined,
      ],
    ]
  }
/>
`;

exports[`Column for Web should render View with class = \`\`.
The following classes should not be displayed: col-xs col-sm col-md col-lg
Test Data: xs=null sm=null md=null lg=null 1`] = `
<View
  data-class=""
  style={
    Array [
      undefined,
      Array [
        undefined,
        undefined,
      ],
    ]
  }
/>
`;

exports[`Column for Web should render View with outer and inner column (parent and child).
Outer column (parent) should have class = \` col-xs-9 col-sm-10 col-md-11 col-lg-12\`, "width": 9, and "margin": 1
Inner column (child) should have class = \` col-xs-1 col-sm-2 col-md-3 col-lg-4\`, and "margin": 0.2
Test Data: Outer_Column { lg={12} md={11} sm={10} xs={9} absolute={true} "margin": 1}
Inner_Column { lg={4} md={3} sm={2} xs={1} absolute={false} "margin": 0.2} 1`] = `
<View
  data-class=" col-xs-9 col-sm-10 col-md-11 col-lg-12"
  style={
    Array [
      undefined,
      Array [
        Array [
          undefined,
          Object {
            "margin": 1,
          },
        ],
        Object {
          "width": 9,
        },
      ],
    ]
  }
>
  <View
    data-class=" col-xs-1 col-sm-2 col-md-3 col-lg-4"
    style={
      Array [
        undefined,
        Array [
          undefined,
          Object {
            "margin": 0.2,
          },
        ],
      ]
    }
  />
</View>
`;

exports[`Column for Web should render View with outer and inner column (parent and child).
Outer column (parent) should have class = \` col-xs-9 col-sm-10 col-md-11 col-lg-12\`, and "margin": 1.
Inner column (child) should have class = \` col-xs-1 col-sm-2 col-md-3 col-lg-4\`, "width": 1, and "margin": 0.2
Test Data: Outer_Column { lg={12} md={11} sm={10} xs={9} absolute={false} "margin": 1}
Inner_Column { lg={4} md={3} sm={2} xs={1} absolute={true} "margin": 0.2} 1`] = `
<View
  data-class=" col-xs-9 col-sm-10 col-md-11 col-lg-12"
  style={
    Array [
      undefined,
      Array [
        undefined,
        Object {
          "margin": 1,
        },
      ],
    ]
  }
>
  <View
    data-class=" col-xs-1 col-sm-2 col-md-3 col-lg-4"
    style={
      Array [
        undefined,
        Array [
          Array [
            undefined,
            Object {
              "margin": 0.2,
            },
          ],
          Object {
            "width": 1,
          },
        ],
      ]
    }
  />
</View>
`;
